File: Level 1/Week01/Day1/kjh1.java
import java.util.*;

class Solution {
  public int solution(int n) {
    boolean[] isPrime = new boolean[n + 1];
    Arrays.fill(isPrime, true);

    for (int i = 2; i <= n; i++) {
      if (isPrime[i] == false)
        continue;

      for (int j = i + i; j <= n; j += i) {
        isPrime[j] = false;
      }
    }

    int primeCount = 0;
    for (int i = 2; i <= n; i++) {
      if (isPrime[i])
        primeCount++;
    }

    return primeCount;
  }
}

File: Level 1/Week01/Day4/kjh1.kt
import java.util.*;

class Solution {
    fun solution(board: Array<IntArray>, moves: IntArray): Int {
        val sections = Array<Stack<Int>>(board[0].size) { Stack<Int>() }
        
        for (secIdx in 0..board[0].size-1) {
            for (i in board.size-1 downTo 0) {
                val doll = board[i][secIdx]
                if (doll == 0) break
                sections[secIdx].push(doll)
            }
        }
        
        var goneDolls = 0
        
        val basket = Stack<Int>()
        for (move in moves) {
            val secIdx = move-1
            if (sections[secIdx].isEmpty()) continue
            
            val pickedDoll = sections[secIdx].pop()
            if (!basket.isEmpty() && basket.peek() == pickedDoll) {
                basket.pop()
                goneDolls += 2
                continue
            }
            basket.push(pickedDoll)
        }
        
        return goneDolls
    }
}